test.sh

```diff
@@ -769,3 +769,11 @@ total_count=$(
   curl -sSfg -u $SK: $HOST/v1/charges?customer=$cus\&created[gt]=1588166306 \
   | grep -oE '"total_count": 6')
 [ -n "$total_count" ]
+
+# check starting_after parameter for the events
+# get the last event id
+event_id=$(curl -sSfg -u $SK: $HOST/v1/events?limit=10000 | grep -oE 'evt_\w+' | tail -n 1)
+
+# after the last event id there should not be more events
+events_none=$(curl -sSfg -u $SK: $HOST/v1/events?starting_after=$event_id | grep -oE '"has_more": false')
+[ -n "$events_none" ]
```

Comment:
This test makes sense only if the list is non empty without `starting_after`.

What do you think of this?
```sh
no_more_events=$(curl -sSfg -u $SK: $HOST/v1/events | grep -oE '"has_more": false')
[ -z "$no_more_events" ]
last_event=$(curl -sSfg -u $SK: $HOST/v1/events?limit=100 | grep -oE 'evt_\w+' | tail -n 1)
no_more_events=$(curl -sSfg -u $SK: $HOST/v1/events?starting_after=$last_event | grep -oE '"has_more": false')
[ -n "$no_more_events" ]
```

Reply:
I don't really get what you doing here.

Reply:
The thing is that the nested event object may have a `has_more` field.

Reply:
> I don't really get what you doing here.

Here is the idea:
\- List all events, check that there are more (`"has_more": false` is not present).
\- List all events *after the last one*, check that there are no more (`"has_more": false` is present).
An extra check is to verify that zero events are returned on the second call (see my example below).

> The thing is that the nested event object may have a `has_more` field.

That's true. So, this means that the initial test you wrote doesn't check anything neither, right?
In this case, you just need to check `has_more` fields that are correctly indented: `grep -oE '^  "has_more": false'`.

Please use this:
```sh
no_more_events=$(curl -sSfg -u $SK: $HOST/v1/events \
                 | grep -oE '^  "has_more": false' || true)
[ -z "$no_more_events" ]
last_event=$(curl -sSfg -u $SK: $HOST/v1/events?limit=100 \
             | grep -oE 'evt_\w+' | tail -n 1)
no_more_events=$(curl -sSfg -u $SK: $HOST/v1/events?starting_after=$last_event \
                 | grep -oE '^  "has_more": false')
[ -n "$no_more_events" ]
zero_events=$(curl -sSfg -u $SK: $HOST/v1/events?starting_after=$last_event \
                 | grep -oE '^  "data": \[\]')
[ -n "$zero_events" ]
```